src/pages/basic_data/institution_admin/edit_user/AreaOfResponsibility.tsx

```diff
@@ -63,14 +61,34 @@ export const AreaOfResponsibility = ({ viewingScopes, updateViewingScopes }: Are
           fullWidth
           data-testid={dataTestId.myInstitutionUsersPage.areaOfResponsibilityField}
           options={options}
+          filterOptions={(options, state) =>
+            options.filter((option) => {
+              const input = state.inputValue.toLowerCase();
+              return (
+                Object.values(option.labels).some((label) => label.toLowerCase().includes(input)) ||
+                option.acronym?.toLowerCase().includes(input)
+              );
+            })
+          }
           getOptionLabel={(option) => getLanguageString(option.labels)}
           renderOption={({ key, ...props }, option) => (
             <OrganizationRenderOption key={option.id} props={props} option={option} />
           )}
           disabled={isSubmitting}
-          onChange={(_, value) => {
+          onChange={async (_, value) => {
             if (value) {
-              updateViewingScopes([...viewingScopes, value.id]);
+              const fetchedUnit = await fetchOrganization(value.id);
+              const allParentIds = getOrganizationHierarchy(fetchedUnit).map((unit) => unit.id);
+              const allChildrenIds = getAllChildOrganizations([value]).map((unit) => unit.id);
+              const organizationIdsToRemove = Array.from(new Set([...allChildrenIds, ...allParentIds]));
+
+              // Remove existing organizations conflicting with the new organization
+              const filteredScopes = viewingScopes.filter(
+                (scope) => !organizationIdsToRemove.includes(scope) && scope !== value?.id
+              );
+
+              const newViewingScopes = Array.from(new Set([...filteredScopes, value.id]));
```

Comment:
Du har vel allerede sjekket for unikhet i `filteredScopes`?

```suggestion
              const newViewingScopes = [...filteredScopes, value.id];
```

Reply:
På de som skal fjernes ha, men tenkte det kunne være greit her også for å unngå at samme id settes to ganger på noe vis. Feks om man legger til en enhet som er der fra før, eller at vi har noe gammel data som har det